JSON parser not working in repl

If possible, please include complete self contained source files (for example by uploading to https://gist.github.com) that exhibit the issue in addition to quoting from them here. The smaller the example the better!

# Steps to Reproduce
```
idris2 -p contrib
Main> :module Language.JSON
Imported module Language.JSON
Main> parse "{}"
>>= (case let (ts, (l, (c, str'))) =
                case case >>= (scan (Pred isSpace) [] (fastUnpack "{}")) (\{_:0} => let (tok', rest) = _ in assert_total (scan Force r2 tok' rest)) of
                     {Just (tok, rest) => let line' = line + cast (countNLs a str tmap acc col line pred tok)
                                          in let col' = getCols a str tmap acc col line pred tok col in Just (MkBounded (fn (fastPack (reverse tok))) False line col line' col', (line', (col', rest)));
                     Nothing => getFirstToken a str tmap acc col line pred ts str} of
                {Just (tok, (line', (col', rest))) => if pred tok then (reverse acc, (line, (col, []))) else assert_total (tokenise pred line' col' (tok :: acc) tmap rest);
                Nothing => (reverse acc, (line, (col, str)))}
          in (ts, (l, (c, fastPack str'))) of
     {(tokens, (_, (_, ""))) => Just (map (tok ?b) tokens);
     _ => Nothing}) (\{bind:0} =>
                       case case case case doParse False (Terminal "Unrecognised input" (\t =>
                                                                                           if == (.kind (.val t)) (JTPunct (Curly Open))
                                                                                             then Just (tokValue k (.text (.val t)))
                                                                                             else Nothing)) (filter (\x => not (ignored x)) bind) of
                                      {Failure com fatal msg ts => Failure com fatal msg ts;
                                      Res com v xs => case assert_total (doParse com (Force next (.val v)) xs) of
                                                      {Failure com' fatal msg ts => Failure com' fatal msg ts;
                                                      Res com' v' xs => Res com' (mergeBounds v v') xs}} of
                                 {Failure com' fatal msg ts => if com' || Delay fatal then Failure com fatal msg ts else assert_total (doParse False Force y xs);
                                 Res _ val xs => Res com val xs} of
                            {Failure _ _ msg ts => Left (Error msg ts);
                            Res _ v rest => Right (.val v, rest)} of
                       {Right (j, []) => Just j;
                       _ => Nothing})
Main>
```

# Expected Behavior
should output `"{}"`

# Observed Behavior
Weird error above

Additionally if I create a program with similar code and execute it I get:
```
Exception: invalid memory reference.  Some debugging context lost
```

OK, something more interesting has come up. I tried to rebuild Idris and the libraries that I use with it by running `make` rather than `make bootstrap`. This would use the idris2 that had been built using `make bootstrap` to be used to build this new idris2. I had cleaned everything up and reinstalled. This causes the failure. I then tried again but this time with the bootstrap build again however it doesn't seem possible to install the idris2 api with the bootstrap build, I get
```
idris2 --install idris2api.ipkg
(toplevel):1:1--1:1:Text.PrettyPrint.Prettyprinter not found
make: *** [install-api] Error 1
```
So basically I've lost whatever version of idris2 I had that worked but I can say that this error appears to occur for a new build of commit `a7cb2745`.

--- a/libs/contrib/Text/Distance/Levenshtein.idr
+++ b/libs/contrib/Text/Distance/Levenshtein.idr
@@ -10,7 +10,7 @@ import Data.List.Extra
 
 ||| Self-evidently correct but O(3 ^ (min mn)) complexity
 spec : String -> String -> Nat
-spec a b = loop (fastUnpack a) (fastUnpack b) where
+spec a b = loop (unpack a) (unpack b) where
 
   loop : List Char -> List Char -> Nat
   loop [] ys = length ys -- deletions

--- a/libs/contrib/Text/Lexer/Core.idr
+++ b/libs/contrib/Text/Lexer/Core.idr
@@ -159,7 +159,7 @@ tokenise pred line col acc tmap str
                Just (tok, rest) =>
                  let line' = line + cast (countNLs tok)
                      col' = getCols tok col in
-                     Just (MkBounded (fn (fastPack (reverse tok))) False (MkBounds line col line' col'),
+                     Just (MkBounded (fn (pack (reverse tok))) False (MkBounds line col line' col'),
                            line', col', rest)
                Nothing => getFirstToken ts str
 
@@ -171,11 +171,11 @@ export
 lex : TokenMap a -> String -> (List (WithBounds a), (Int, Int, String))
 lex tmap str
     = let (ts, (l, c, str')) = tokenise (const False) 0 0 [] tmap (unpack str) in
-          (ts, (l, c, fastPack str'))
+          (ts, (l, c, pack str'))
 
 export
 lexTo : (a -> Bool) ->
         TokenMap a -> String -> (List (WithBounds a), (Int, Int, String))
 lexTo pred tmap str
     = let (ts, (l, c, str')) = tokenise pred 0 0 [] tmap (unpack str) in
-          (ts, (l, c, fastPack str'))
+          (ts, (l, c, pack str'))

--- a/libs/contrib/Text/Lexer/Tokenizer.idr
+++ b/libs/contrib/Text/Lexer/Tokenizer.idr
@@ -110,7 +110,7 @@ tokenise reject tokenizer line col acc str
             | _ => Nothing
           line' = line + cast (countNLs token)
           col' = getCols token col
-          tokenStr = fastPack $ reverse token
+          tokenStr = pack $ reverse token
        in pure (tokenStr, line', col', rest)
 
     getFirstMatch : Tokenizer a -> List Char ->
@@ -149,8 +149,8 @@ lexTo : Lexer ->
         (List (WithBounds a), (StopReason, Int, Int, String))
 lexTo reject tokenizer str
     = let (ts, reason, (l, c, str')) =
-              tokenise reject tokenizer 0 0 [] (fastUnpack str) in
-          (ts, reason, (l, c, fastPack str'))
+              tokenise reject tokenizer 0 0 [] (unpack str) in
+          (ts, reason, (l, c, pack str'))
 
 ||| Given a tokenizer and an input string, return a list of recognised tokens,
 ||| and the line, column, and remainder of the input at the first point in the string

--- a/libs/contrib/Text/Literate.idr
+++ b/libs/contrib/Text/Literate.idr
@@ -78,7 +78,7 @@ namespace Compat
 
 ||| Merge the tokens into a single source file.
 reduce : List (WithBounds Token) -> List String -> String
-reduce [] acc = fastConcat (reverse acc)
+reduce [] acc = concat (reverse acc)
 reduce (MkBounded (Any x) _ _ :: rest) acc =
   -- newline will always be tokenized as a single token
   if x == "\n"

--- a/libs/contrib/Text/PrettyPrint/Prettyprinter/Render/HTML.idr
+++ b/libs/contrib/Text/PrettyPrint/Prettyprinter/Render/HTML.idr
@@ -6,7 +6,7 @@ import Data.String
 
 export
 htmlEscape : String -> String
-htmlEscape s = fastConcat $ reverse $ go [] s
+htmlEscape s = concat $ reverse $ go [] s
   where
     isSafe : Char -> Bool
     isSafe '"' = False

